x/auth/ante/sigverify.go

```diff
@@ -211,7 +227,7 @@ func (svd SigVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 		// the SignatureV2 struct (it's only in the SignDoc). In this case, we
 		// cannot check sequence directly, and must do it via signature
 		// verification.
-		if !sig.SkipSequenceCheck {
+		if !OnlyLegacyAminoSigners(sig.Data) {
```

Comment:
let's say that that the `sig.Data` is multi-sig with 2 signatures: one amino, and one direct.
How do you make sure that the `amino-sig` is signed with a correct sequence number?

Reply:
In this case you describe, the VerifySignature call below (line 237/253) will fail if there is an incorrect sequence number.

Reply:
hmm, in that case we don't need this `if` block - it's only an early optimization?

Reply:
It's this `if` block that allows us to give an explicit error to mismatch sequence numbers. When erroring in signature verification, all we know is that the signature is not valid (not whether the account sequence was the actual problem). In `SIGN_MODE_DIRECT` we want to provide explicit feedback as to whether or not the sequence number is valid.

Reply:
This `if` statement is actually needed for replay protection for SIGN_MODE_DIRECT, so not only for feedback.

In SIGN_MODE_DIRECT, signer claims his Sequence is `n`, and signs that. The signature verification only checks that the signature matches the claim. The check "your on-chain Sequence matches `n`" is done in this `if` block.

In SIGN_MODE_LEGACY, both checks are done in signature verification.

Reply:
> In SIGN_MODE_DIRECT, signer claims his Sequence is n, and signs that. The signature verification only checks that the signature matches the claim. The check "your on-chain Sequence matches n" is done in this if block.

So this was my main concern. In the example I described (multisig with 2 signatures: s1: legacy, s2: direct), we won't do this check for `s2` -- in the loop below, the `VerifySignature` for `s2` won't check the Sequence number. Correct?

Reply:
we verified together and it looks good.